lib/phoenix_live_view/lifecycle.ex

```diff
@@ -156,15 +156,27 @@ defmodule Phoenix.LiveView.Lifecycle do
         {:halt, %Socket{redirected: nil}} ->
           raise_halt_without_redirect!(hook)
 
+        {:halt, %Socket{redirected: nil}, _opts} ->
+          raise_halt_without_redirect!(hook)
+
         {:cont, %Socket{redirected: to}} when not is_nil(to) ->
           raise_continue_with_redirect!(hook)
 
+        {:cont, %Socket{redirected: to}, _opts} when not is_nil(to) ->
+          raise_continue_with_redirect!(hook)
+
         ok ->
-          ok
+          handle_on_mount_opts!(ok)
```

Comment:
because this is returning a tuple I presume it shouldn't be `!` at the end but cribbed from the implementation in `Utils` until told otherwise